dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/AuthoritiesController.java

```diff
@@ -79,14 +101,35 @@ public Map<String, List<Map<String, String>>> getAuthorities(HttpServletResponse
     return singletonMap("systemAuthorities", entries);
   }
 
+  public Collection<String> getAppAuthorities() {
+    Set<String> authorities = new HashSet<>();
+    appManager.getApps(null).stream()
+        .filter(app -> !StringUtils.isEmpty(app.getShortName()) && !app.isBundled())
+        .forEach(
+            app -> {
+              authorities.add(app.getSeeAppAuthority());
+              authorities.addAll(app.getAuthorities());
+            });
+    authorities.add(AndroidSettingsApp.AUTHORITY);
+    return authorities;
+  }
+
+  private List<String> getBundledAppsAuthorities() {
+    List<String> authorities = new ArrayList<>();
+    Set<String> bundledApps = AppManager.BUNDLED_APPS;
+    for (String app : bundledApps) {
+      String key = "M_dhis-web-" + app;
```

Comment:
I wonder if this logic should be the same or based on the same code as the one in `App#getSeeAppAuthority`

Reply:
Yep, best to get the authority from the App itself (if possible) rather than create it outside of the App.

Reply:
I think we should possibly refactor the AppManager code first. I'm not sure that we want to continue using a hard-coded static list of bundled apps.